src/exoticatechnologies/hullmods/AlphaSubcoreHM.kt

```diff
@@ -7,37 +7,108 @@ import com.fs.starfarer.api.combat.ShipAPI.HullSize
+            }
         } else {
             stats.removeListener(listener)
             listenerAddedMap[stats] = false
         }
+
+        AnonymousLogger.log("listenerAddedMap: ${listenerAddedMap}", "AlphaSubcoreHM")
+    }
+
+    override fun removeEffectsBeforeShipCreation(hullSize: HullSize, stats: MutableShipStatsAPI, id: String) {
+        AnonymousLogger.log("--> removeEffectsBeforeShipCreation()\thullSize: ${hullSize}, stats: ${stats}, id: ${id}", "AlphaSubcoreHM")
+        val fleetMember = stats.fleetMember
+        AnonymousLogger.log("removeEffectsBeforeShipCreation()\tfleetMember: ${fleetMember}", "AlphaSubcoreHM")
+//        val listenerAdded = listenerAddedMap[stats]
+//        AnonymousLogger.log("removeEffectsBeforeShipCreation()\tlistenerAdded: ${listenerAdded}", "AlphaSubcoreHM")
+        AnonymousLogger.log("removeEffectsBeforeShipCreation()\tlistenerAdded: ${listenerAddedMap[stats]}", "AlphaSubcoreHM")
+        stats.removeListenerOfClass(OPCostListener::class.java)
+        if (stats is ShipAPI) {
+            AnonymousLogger.log("removeEffectsBeforeShipCreation()\tstats is ShipAPI !!!", "AlphaSubcoreHM")
+            val ship = stats as ShipAPI
+            val entity = stats.entity
+            AnonymousLogger.log("removeEffectsBeforeShipCreation()\tis stats.entity a ShipAPI ? ${entity is ShipAPI}", "AlphaSubcoreHM")
+//            val modules = getAllShipSections(ship)    //TODO
+            val modules = ship.childModulesCopy
+            AnonymousLogger.log("removeEffectsBeforeShipCreation()\tchildModules size: ${modules.size}", "AlphaSubcoreHM")
+            for (module in modules) {
+//                removeEffectsBeforeShipCreation(module.hullSize, module, id)
+//                removeEffectsBeforeShipCreation(module.hullSize, module.mutableStats, id)    //TODO
+                removeEffectsBeforeShipCreation(module.hullSize, module as MutableShipStatsAPI, id)    //TODO
+            }
+            AnonymousLogger.log("removeEffectsBeforeShipCreation()\tdone with child modules, lets try the ExoticDataHandler ...", "AlphaSubcoreHM")
+            // Since childModulesCopy doesnt quite seem to be working
+            val allKeys = HullmodExoticHandler.grabAllKeysForParticularFleetMember(fleetMember)
+            for (key in allKeys) {
+                val exoticHandlerDataOptional = HullmodExoticHandler.getDataForKey(key)
+                if (exoticHandlerDataOptional.isPresent()) {
+                    val exoticHandlerData = exoticHandlerDataOptional.get()
+                    for (variant in exoticHandlerData.listOfVariantsWeInstalledOn) {
+                        val variantHullSize = variant.hullSpec.hullSize
+                        removeEffectsBeforeShipCreation(variantHullSize, variant.statsForOpCosts, id)
+                    }
+                }
+                //TODO keep track of the keys we 'removed' so we can actually remove them from the HullmodExoticHandler's lookupMap
+            }
```

Comment:
this should literally be extracted as `HullmodExoticHandler.removeHullmodExoticFromFleetMember(exoticHullmod: ExoticHullmod, fleetMember: FleetMemberAPI)`

Reply:
it's done even better. resolving.